src/utils/helpers.js

```diff
@@ -65,4 +61,29 @@ export default class Helper {
 
     return token;
   }
+
+  /**
+   * Method generate token
+   * @param {object} profile -user social profile
+   * @returns {object} - user details
+   */
+  static getUserSocialDetails(profile) {
+    const { provider, _json: data } = profile;
+    const user = { password: crypto.randomBytes(15).toString('hex') };
+    if (provider === 'google') {
+      user.email = data.email;
+      user.image = data.picture;
+      user.social_id = data.sub;
+      user.first_name = data.given_name;
+      user.last_name = data.family_name;
+    } else {
+      user.social_id = data.id;
+      user.email = data.email;
+      user.first_name = data.first_name;
+      user.last_name = data.last_name;
+      user.middle_name = data.middle_name;
+      user.image = data.picture.data.url;
+    }
```

Comment:
You can apply to destructuring here from line 74 to 86
{ email, image, social_id, ... } = user;

Reply:
Am the one construction the user object from the profile sent by Google or Facebook, because I want to reduce the numbers of code handling the user profiles details that why I created this function.